lucene/facet/src/java/org/apache/lucene/facet/DrillSideways.java

```diff
@@ -390,18 +398,44 @@ protected boolean scoreSubDocsAtOnce() {
     return false;
   }
 
-  /** Result of a drill sideways search, including the {@link Facets} and {@link TopDocs}. */
+  /**
+   * Result of a drill sideways search, including the {@link Facets} and {@link TopDocs} {@link
+   * FacetsCollector} for drill down and drill sideways passed to {@link #buildFacetsResult} to
+   * compute {@link Facets} are also exposed here as an alternative to subclassing DrillSideways to
+   * obtain them
+   */
   public static class DrillSidewaysResult {
     /** Combined drill down and sideways results. */
     public final Facets facets;
 
     /** Hits. */
     public final TopDocs hits;
 
+    /** FacetsCollector for drill down. */
+    public final FacetsCollector drillDowns;
+
+    /**
+     * Facets Collector for drill sideways Array of FacetsCollector maps directly to Array of
+     * String[] drillSidewaysDims i.e. ith drillSidewaysDims String maps to ith drillSideways
+     * FacetsCollector
+     */
```

Comment:
I think some users might have difficulty understanding this documentation. Maybe something like:

```suggestion
    /**
     * FacetsCollectors populated for each drill sideways dimension. Each collector exposes
     * the hits that match on all DrillDownQuery dimensions, but treating their corresponding
     * sideways dimension as optional. This array provides a FacetsCollector for each drill
     * down dimension present in the original DrillDownQuery, and the associated dimension
     * for each FacetsCollector can be determined using the parallel `drillSidewaysDims`
     * array. Useful for advanced use-cases that want to compute Facets results separate    
     * from the provided Facets in this result.
     */
```